include/onnxruntime/core/framework/allocator.h

```diff
@@ -101,7 +101,13 @@ class IAllocator {
   const OrtMemoryInfo& Info() const { return memory_info_; };
 
   // Each implementation of IAllocator can override and provide their own implementation
-  virtual void GetStats(AllocatorStats* /*stats*/) { return; }
+  virtual void GetStats(AllocatorStats* /*stats*/) {
+#ifdef ORT_NO_RTTI
+    ORT_NOT_IMPLEMENTED(__FUNCTION__, " is not supported by this allocator");
+#else
+    ORT_NOT_IMPLEMENTED(__FUNCTION__, " is not supported by allocator of type ", typeid(*this).name());
```

Comment:
When RTTI is not enabled, the process would abort. We don't want this to happen. I do not have a suggestion now. Let's thing about it.